spec/requests/api/providers_spec.rb

```diff
@@ -438,6 +455,19 @@ def token(connection)
     end
   end
 
+  describe "custom attributes assignment" do
+    it "custom attributes assignment" do
+      api_basic_authorize collection_action_identifier(:providers, :update_custom_attributes)
+      p1 = FactoryGirl.create(:ext_management_system, :name => "provider")
+      run_post(providers_url, gen_request(:update_custom_attributes, custom_attributes_request))
+      expect(p1.custom_attributes.count).to eq(3)
+      expect(CustomAttribute.find_by_name("experation_date").serialized_value.class).to eq(Date)
```

Comment:
Could this test possibly leverage the response body to verify what changed? This is what forms part of our contract with the end user - they will never make decisions based on the back-end implementation, which they will never see.

Minor nit, but can you add a blank line between each phase of the test for longer tests like this (i.e. setup, execute, verify, teardown)